src/prefect/tasks.py

```diff
@@ -1539,6 +1612,11 @@ def my_flow():
                 "`task.map()` is not currently supported by `flow.visualize()`"
             )
 
+        # Use stored dependencies if available, then clear them
+        if self._upstream_dependencies is not None:
+            wait_for = self._upstream_dependencies
+            self._upstream_dependencies = None
```

Comment:
Do we want to clear the dependencies on each call? What if you have many task calls that all depend on the same upstream dependencies? Do you need to call `.depends_on` each time?